components/brave_shields/browser/ad_block_engine.cc

```diff
@@ -191,59 +188,74 @@ void AdBlockBaseService::EnableTag(const std::string& tag, bool enabled) {
   }
 }
 
-void AdBlockBaseService::AddResources(const std::string& resources) {
+void AdBlockEngine::AddResources(const std::string& resources) {
   if (BrowserThread::CurrentlyOn(BrowserThread::UI)) {
     GetTaskRunner()->PostTask(
-        FROM_HERE, base::BindOnce(&AdBlockBaseService::AddResources,
+        FROM_HERE, base::BindOnce(&AdBlockEngine::AddResources,
                                   base::Unretained(this), resources));
     return;
   }
 
   ad_block_client_->addResources(resources);
-  resources_ = resources;
 }
 
-bool AdBlockBaseService::TagExists(const std::string& tag) {
+bool AdBlockEngine::TagExists(const std::string& tag) {
   return std::find(tags_.begin(), tags_.end(), tag) != tags_.end();
 }
 
-absl::optional<base::Value> AdBlockBaseService::UrlCosmeticResources(
+absl::optional<base::Value> AdBlockEngine::UrlCosmeticResources(
     const std::string& url) {
-  // if (!IsInitialized())
-  //   return;
-
   DCHECK(GetTaskRunner()->RunsTasksInCurrentSequence());
   return base::JSONReader::Read(ad_block_client_->urlCosmeticResources(url));
 }
 
-absl::optional<base::Value> AdBlockBaseService::HiddenClassIdSelectors(
+absl::optional<base::Value> AdBlockEngine::HiddenClassIdSelectors(
     const std::vector<std::string>& classes,
     const std::vector<std::string>& ids,
     const std::vector<std::string>& exceptions) {
-  // if (!IsInitialized())
-  //   return;
-
   DCHECK(GetTaskRunner()->RunsTasksInCurrentSequence());
   return base::JSONReader::Read(
       ad_block_client_->hiddenClassIdSelectors(classes, ids, exceptions));
 }
 
-void AdBlockBaseService::GetDATFileData(const base::FilePath& dat_file_path,
-                                        bool deserialize,
-                                        base::OnceClosure callback) {
+void AdBlockEngine::OnInitialListLoad(bool deserialize,
+                                      const DATFileDataBuffer& dat_buf) {
+  if (deserialize) {
```

Comment:
I thought this was handled by the engine?

Reply:
> I thought this was handled by the engine?

The engine handles different serialized formats automatically, but currently doesn't accept list source text through the same codepath.

> also why is the initial load different from normal component update reloads?

It shouldn't be - this method just delegates to either `OnDATLoaded` or `OnListSourceLoaded`, which are also used for subsequent reloads.

Although I'm realizing now that `AdBlockEngine::GetDATFileData` and `AdBlockEngine::OnGetDATFileData` are now dead code and can be removed.